### PR TITLE
Asset deck

### PR DESCRIPTION
**Pull request by Eddasol**

### CODE REVIEW

**Comment by Eddasol** [comment]
Dependant on https://github.com/equinor/isar/pull/437

File: backend/api/Database/Models/AssetDeck.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+using Microsoft.EntityFrameworkCore;
+
+#nullable disable
```
Comment (eivindsjovold) [inline backend/api/Database/Models/AssetDeck.cs:6]: Is this needed when all properties are required?

File: backend/api/Database/Models/AssetDeck.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        [Key]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public string Id { get; set; }
+
+        [Required]
+        [MaxLength(200)]
+        public string DeckName { get; set; }
+
+        [Required]
+        public Pose DefaultPose { get; set; }
```
Comment (eivindsjovold) [inline backend/api/Database/Models/AssetDeck.cs:21]: Should Default maybe be named DefaultLocalizationPose?

File: backend/api/Services/RobotService.cs [hunk 3]
```diff
@@ -48,6 +48,20 @@ public async Task<Robot> Create(CreateRobotQuery newRobot)
                 videoStreams.Add(videoStream);
             }
 
+            var availableDecks = new List<AssetDeck>();
+            foreach (var assetDeckQuery in newRobot.AvailableDecks)
```
Comment (eivindsjovold) [inline backend/api/Services/RobotService.cs:52]: Why assetDeckQuery? 

```suggestion
            foreach (var assetDecks in newRobot.AvailableDecks)
```

File: backend/api/Services/RobotService.cs [hunk 4]
```diff
@@ -48,6 +48,20 @@ public async Task<Robot> Create(CreateRobotQuery newRobot)
...
+            {
+                var defaultPose = new Pose();
```
Comment (eivindsjovold) [inline backend/api/Services/RobotService.cs:54]: why empty pose?

File: backend/api/Services/RobotService.cs [hunk 5]
```diff
@@ -48,6 +48,20 @@ public async Task<Robot> Create(CreateRobotQuery newRobot)
...
+                assetDeckQuery.DefaultPose.CopyIsarPoseToRobotPose(defaultPose);
+
+                var assetDeck = new AssetDeck
+                {
+                    DeckName = assetDeckQuery.DeckName,
+                    DefaultPose = defaultPose
+                };
+                availableDecks.Add(assetDeck);
+            }
+
```
Comment (eivindsjovold) [inline backend/api/Services/RobotService.cs:64]: is it smarter to write a function that takes in AvailableDecks and converts to a list of asset decks in the constructor?

File: frontend/src/models/AssetDeck.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { Pose } from './Pose'
+
+export interface AssetDeck {
+    id: string
+    deckName: string
+    defaultPose: Pose
```
Comment (eivindsjovold) [inline frontend/src/models/AssetDeck.ts:6]: defaultLocalizationPose?

File: backend/api/Database/Models/AssetDeck.cs (hunk 1 above)
Comment (Eddasol) [reply backend/api/Database/Models/AssetDeck.cs:7]: The constructor functions still need it

File: backend/api/Controllers/AssetDeckController.cs [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<AssetDeck>> Create(
+        [FromBody] CreateAssetDeckQuery assetDeck
+    )
+    {
+        _logger.LogInformation("Creating new asset deck");
+        try
+        {
+            var newAssetDeck = await _assetDeckService.Create(assetDeck);
+            _logger.LogInformation("Succesfully created new asset deck");
```
Comment (GodVenn) [inline backend/api/Controllers/AssetDeckController.cs:89]:
```suggestion
            _logger.LogInformation("Succesfully created new asset deck with id '{assetDeckId}'", newAssetDeck.Id);
```

Comment (GodVenn) [reply backend/api/Controllers/AssetDeckController.cs:89]: Include Id in log message

**Review by GodVenn — COMMENTED** [review]
I think the asset decks should consistently be called asset decks instead of available decks some places, otherwise looks nice

File: backend/api/Controllers/AssetDeckController.cs [hunk 8]
```diff
@@ -0,0 +1,117 @@
+﻿using Api.Controllers.Models;
+using Api.Database.Models;
+using Api.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Api.Controllers;
+
+[ApiController]
+[Route("available-decks")]
```
Comment (GodVenn) [inline backend/api/Controllers/AssetDeckController.cs:9]:
```suggestion
[Route("asset-decks")]
```

**Review by GodVenn — CHANGES_REQUESTED** [review]
LGTM other than one forgotten available decks

**Pull request merged**